app/src/main/java/kth/jjve/moveable/MainActivity.java

```diff
@@ -427,45 +432,49 @@ public void savingCancelled() {
     @Override
     public void onSensorChanged(SensorEvent event) {
         if (event.sensor.getType()==Sensor.TYPE_ACCELEROMETER){
-            ax=event.values[0];
-            ay=event.values[1];
-            az=event.values[2];
+            float ax=event.values[0];
+            float ay=event.values[1];
+            float az=event.values[2];
 
             // rotation from acceleration data
-            roll_x = Math.toDegrees(Math.atan(ay / (Math.sqrt(Math.pow(ax, 2) + Math.pow(ax, 2)))));
-            pitch_y = Math.toDegrees(Math.atan(ax / (Math.sqrt(Math.pow(ay, 2) + Math.pow(ax, 2)))));
-            yaw_z = Math.toDegrees(Math.atan( (Math.sqrt(Math.pow(ax, 2) + Math.pow(ay, 2)) / az )));
+            pitchRollYaw = DataProcess.rotFromAcc(ax, ay, az);
```

Comment:
I don't think we need to display all right? So I would only return one and depending on orientation of the screen

Reply:
Yeah that's the goal, but since I haven't validated the angles yet, I will need them